### DISCUSSION TITLE
Doris命中物化视图导致查询数据异常

### DISCUSSION THREAD

**Discussion by Jrebel-i** [apache/doris#35015] [A - General / Q&A]
doris version：2.1.2
物化视图（没有sum,count等聚合操作）：create materialized view app_hotel_supplier_product_d0_d1_comparison_d_mv_2 as
select day_num
,supplier
,supplier_code
,country_code
,supplier_hotel_id
from app.app_hotel_supplier_product_d0_d1_comparison_d
group by day_num
,supplier
,supplier_code
,country_code
,supplier_hotel_id
![Snipaste_2024-05-17_16-32-01](https://github.com/apache/doris/assets/62320242/4b86024b-69d1-4a3e-a2d8-9ad410cfc16c)

**Comment by yagagagaga** [comment]
Can you provide table ddl and query?

**Reply by Jrebel-i** [reply]
我提供了复现步骤，在2.1.2，2.1.3版本doris运行

**Comment by Jrebel-i** [comment]
> app.app_hotel_supplier_product_d0_d1_comparison_d

CREATE TABLE `app_hotel_supplier_product_d0_d1_comparison_d` (
  `supplier` VARCHAR(100) NULL COMMENT '供应商',
  `supplier_code` VARCHAR(100) NULL COMMENT '供应账号',
  `supplier_hotel_id` VARCHAR(100) NULL COMMENT '供应商酒店ID',
  `master_hotel_id` BIGINT NULL COMMENT '主酒店ID',
  `supplier_room_id` VARCHAR(8000) NULL COMMENT '供应商房间ID',
  `master_room_id` BIGINT NULL COMMENT '主房间ID',
  `country_code` VARCHAR(100) NULL COMMENT '国家代码',
  `is_online` VARCHAR(10) NULL COMMENT '是否在线',
  `is_mapping` VARCHAR(10) NULL COMMENT '是否映射',
  `is_empty` VARCHAR(10) NULL COMMENT '是否为空',
  `check_in_date` VARCHAR(20) NULL COMMENT '入住日期',
  `check_out_date` VARCHAR(20) NULL COMMENT '退房日期',
  `min_price_d0` DECIMAL(10, 2) NULL COMMENT '当天最低价格',
  `min_price_d1` DECIMAL(10, 2) NULL COMMENT '前一天最低价格',
  `day_num` DATE NULL COMMENT '分区日期'
) ENGINE=OLAP
DUPLICATE KEY(`supplier`)
COMMENT '供应相同产品前后两天价格对比'
PARTITION BY RANGE(`day_num`)
(PARTITION p20240512 VALUES [('2024-05-12'), ('2024-05-13')),
PARTITION p20240513 VALUES [('2024-05-13'), ('2024-05-14')),
PARTITION p20240514 VALUES [('2024-05-14'), ('2024-05-15')),
PARTITION p20240515 VALUES [('2024-05-15'), ('2024-05-16')),
PARTITION p20240516 VALUES [('2024-05-16'), ('2024-05-17')),
PARTITION p20240517 VALUES [('2024-05-17'), ('2024-05-18')),
PARTITION p20240518 VALUES [('2024-05-18'), ('2024-05-19')),
PARTITION p20240519 VALUES [('2024-05-19'), ('2024-05-20')),
PARTITION p20240520 VALUES [('2024-05-20'), ('2024-05-21')),
PARTITION p20240521 VALUES [('2024-05-21'), ('2024-05-22')),
PARTITION p20240522 VALUES [('2024-05-22'), ('2024-05-23')),
PARTITION p20240523 VALUES [('2024-05-23'), ('2024-05-24')),
PARTITION p20240524 VALUES [('2024-05-24'), ('2024-05-25')))
DISTRIBUTED BY HASH(`supplier`, `supplier_hotel_id`, `master_hotel_id`, `supplier_room_id`, `master_room_id`) BUCKETS AUTO
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"min_load_replica_num" = "-1",
"is_being_synced" = "false",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-10",
"dynamic_partition.end" = "2",
"dynamic_partition.prefix" = "p",
"dynamic_partition.replication_allocation" = "tag.location.default: 1",
"dynamic_partition.buckets" = "3",
"dynamic_partition.create_history_partition" = "true",
"dynamic_partition.history_partition_num" = "-1",
"dynamic_partition.hot_partition_num" = "0",
"dynamic_partition.reserved_history_periods" = "NULL",
"dynamic_partition.storage_policy" = "",
"dynamic_partition.storage_medium" = "HDD",
"storage_format" = "V2",
"inverted_index_storage_format" = "DEFAULT",
"compression" = "ZSTD",
"light_schema_change" = "true",
"disable_auto_compaction" = "false",
"enable_single_replica_compaction" = "false",
"group_commit_interval_ms" = "10000",
"group_commit_data_bytes" = "134217728"
);

**Comment by Jrebel-i** [comment]
![image](https://github.com/apache/doris/assets/62320242/46479eb6-3f04-485e-9c33-05a7d19f0a75)
CREATE TABLE IF NOT EXISTS `test`.`test_mv_bug`
(
    `id` int not null,
    `name` varchar(100) not null ,
    `age` int not null ,
    `sex` int null,
    `day_num` DATE
) ENGINE=OLAP
DUPLICATE KEY(id)
COMMENT "测试物化视图bug"
PARTITION BY RANGE(day_num)()
DISTRIBUTED BY HASH(id)
PROPERTIES ( 
    "replication_allocation" = "tag.location.default: 1",
    "dynamic_partition.enable" = "true",
    "dynamic_partition.time_unit" = "DAY",
    "dynamic_partition.prefix" = "p",
    "dynamic_partition.buckets" = "3",
    "dynamic_partition.create_history_partition" = "true",
    "dynamic_partition.history_partition_num" = "5",
    "dynamic_partition.start" = "-4",
    "dynamic_partition.end" = "1",
    "storage_format" = "V2"
);
-- 创建物化视图
create materialized view test_mv_bug_mv_ss_2 as
select day_num,age,sex
from `test`.`test_mv_bug`
GROUP BY day_num,age,sex;

select count(*) from `test`.`test_mv_bug` where day_num='2024-05-20';
